zokrates_core/src/ir/expression.rs

```diff
@@ -69,8 +69,9 @@ impl<T: Field> LinComb<T> {
 
     pub fn try_summand(&self) -> Option<(&FlatVariable, &T)> {
         if self.0.len() == 1 {
-            return self.0.iter().next();
+            return self.0.first().map(|(id, k)| (id, k))
```

Comment:
Am I right to guess that `map` was added to turn `&(_, _)` into `(&_, &_)` ?
Maybe `Option<&(FlatVariable, T)>` is fine as a return type for `try_summand` in which case this would not be needed?
In any case, would be cool to clarify here :)

Reply:
Yes, `map` is added to deref. references inside a tuple. It would be more clear to change return type instead, I'll try to do so.